pkgs/applications/editors/rstudio/default.nix

```diff
@@ -61,17 +67,24 @@ let
   rsconnectSrc = fetchFromGitHub {
     owner = "rstudio";
     repo = "rsconnect";
-    rev = "e287b586e7da03105de3faa8774c63f08984eb3c";
-    sha256 = "sha256-ULyWdSgGPSAwMt0t4QPuzeUE6Bo6IJh+5BMgW1bFN+Y=";
+    rev = "db2bcba0d82698aca6bc6aff4f639cc531060b4b";
+    sha256 = "sha256-Lv3WBTb0oP0ibP3AB0v3+E+i2DJoADMZ6Ysfxnk3ZvE=";
+  };
+
+  quartoSrc = fetchFromGitHub {
+    owner = "quarto-dev";
+    repo = "quarto";
+    rev = "a92906bbd004e3d8abc39b5f59346ca2a091d1e8";
+    sha256 = "sha256-ZpqOxwpqrFqwaR+SkEki9aZhTzsRLh9Rd1tYCC9uukk=";
   };
 
   panmirrorModules = mkYarnModules rec {
     inherit pname version;
     packageJSON = ./package.json;
-    yarnLock = "${src}/src/gwt/panmirror/src/editor/yarn.lock";
+    yarnLock = ./yarn.lock;
```

Comment:
Upstream deleted the panmirror folder, and put yarn.lock in gitignore. I think they manage that file locally, and it disappeared form the git repo. Someone (@nviets ?) placed it in the nixpkgs repo instead.

Reply:
Yeah, that was me. I was fumbling through this yarn part, so you're welcome to do whatever works. There is some discussion on that here: https://github.com/NixOS/nixpkgs/pull/245269.

Reply:
Where did the lock file come from then? This should point to that location to avoid bundling this into nixpkgs.

Reply:
may be simply a copy of the lock file from before? @nviets do you remember how you got it?

Reply:
There's actually a node.js-related error, although it doesn't seem to be a problem. The OCD in me would still like it to go away. If anybody has ideas how to fix it, please don't hold back.

     [exec] Error: Cannot find module '/nix/store/gv66dg8gjckw42m0iwnyh9zx440da9bl-source/apps/panmirror/fuse'
     [exec]     at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
     [exec]     at Module._load (node:internal/modules/cjs/loader:922:27)
     [exec]     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
     [exec]     at node:internal/main/run_main_module:23:47 {
     [exec]   code: 'MODULE_NOT_FOUND',
     [exec]   requireStack: []
     [exec] }
     [exec] 
     [exec] Node.js v18.17.1
     [exec] Result: 1

Reply:
See my original PR for how to get the lock file. Someone explains there

Reply:
I think this whole panmirror bit is broken. The modules is now part of the `quarto` source and in principle the following should work:

```
  panmirrorModules = mkYarnModules rec {
    pname = "panmirror";
    version = "dev";
    packageJSON = "${quartoSrc}/apps/panmirror/package.json";
    yarnLock = "${quartoSrc}/yarn.lock";
    offlineCache = fetchYarnDeps {
      inherit yarnLock;
      hash = "sha256-dykyuVoMyz0aTZkXmspfPL9yMPGJQYGGZ+Y8MMM3cpQ=";
    };
  };
```

It doesn't though, fails because it can't find the `editor` module which is also part of the quarto source. I don't know enough about the yarn/node to progress much further at the moment.